homeassistant/components/zha/core/gateway.py

```diff
@@ -166,6 +177,25 @@ def __init__(self, hass, config, config_entry):
         )
         async_dispatcher_send(self._hass, SIGNAL_ADD_ENTITIES)
 
+    @staticmethod
+    async def async_options_updated(hass, config_entry):
+        """Handle the update signal from updating the config entry."""
+        gateway = hass.data[DATA_ZHA][DATA_ZHA_GATEWAY]
+        await gateway.async_update_options(config_entry)
+
+    async def async_update_options(self, config_entry):
+        """Triggered by config entry options updates."""
+        _LOGGER.debug(
+            "ZHA config entry options updated from: %s to: %s",
+            self._config_entry.options,
+            config_entry.options,
+        )
+        self._config_entry = config_entry
```

Comment:
This is not needed, the data of your config entry has been updated prior to async_options_updated was called